libarchive/archive_string.c

```diff
@@ -418,6 +418,8 @@ default_iconv_charset(const char *charset) {
 	return locale_charset();
 #elif HAVE_NL_LANGINFO
 	return nl_langinfo(CODESET);
+#elif defined(__ANDROID__)
```

Comment:
```suggestion
#elif defined(__BIONIC__)
```
As it's possible to use bionic outside of Android.

Reply:
Thanks, updated and rebased the branch.